packages/mongo-nav/src/mongo-menu/MongoMenu.tsx

```diff
@@ -0,0 +1,309 @@
+  return (
+    <div
+      className={cx(productLinkStyle, {
+        [activeProductLinkStyle]: isActive,
+      })}
+    >
+      {`${product}.mongodb.com`}
+      <Icon size="small" glyph="CaretRight" className={productLinkIconStyle} />
+    </div>
+  );
+}
+
+interface MongoMenuProps {
+  /**
+   * Object that contains information about the active user. {firstName: 'string', lastName: 'string', email: 'string'}
+   */
+  account: AccountInterface;
+
+  /**
+   * MongoDB product that is currently active: ['cloud', 'university', 'support'].
+   */
+  activeProduct: Product;
+
+  /**
+   * Callback invoked after the user clicks log out.
+   */
+  onLogout?: React.MouseEventHandler;
+
+  /**
+   * Callback invoked after the user clicks a product.
+   */
+  onProductChange?: React.MouseEventHandler;
+
+  urls: URLSInterface;
+}
+
+function MongoMenu({
+  account: { firstName, lastName, email, openInvitations },
+  activeProduct,
+  onLogout = () => {},
+  onProductChange = () => {},
+  urls,
+}: MongoMenuProps) {
+  const [open, setOpen] = useState(false);
+
+  const name = `${firstName} ${lastName}`;
+
+  const isAccount = activeProduct === 'account';
+  const cloudProducts = ['cloud', 'stitch', 'charts'];
+  const isCloud = cloudProducts.includes(activeProduct);
+  const isSupport = activeProduct === 'support';
+  const isUniversity = activeProduct === 'university';
+
+  const sharedProps = {
+    target: '_blank',
+    rel: 'noopener noreferrer',
+    onClick: (e: React.MouseEvent) => {
+      onProductChange(e);
+      setOpen(false);
+    },
+  };
+
+  return (
+    <div className={triggerWrapper}>
+      <MongoMenuTrigger open={open} name={firstName} setOpen={setOpen} />
+
+      <Menu open={open} setOpen={setOpen} className={menuStyle}>
+        <div className={headerStyle}>
+          <div className={logoMarkBackground}>
+            <LogoMark height={30} />
+          </div>
+
+          <h3 className={cx(nameStyle, truncate)}>{name}</h3>
+
+          <p className={cx(descriptionStyle, truncate)}>{email}</p>
+
+          <FocusableMenuItem>
+            <Button
+              href={
+                isAccount ? undefined : urls?.mongoMenu?.account?.accountURL
+              }
+              disabled={isAccount}
+              as={isAccount ? 'button' : 'a'}
+            >
+              Manage your MongoDB Account
+            </Button>
+          </FocusableMenuItem>
+        </div>
+        <MenuSeparator />
+
+        <SubMenu
+          {...subMenuContainer.prop}
+          {...sharedProps}
+          active={isCloud}
+          href="https://cloud.mongodb.com"
+          description={<Description isActive={isCloud} product="cloud" />}
+          title="Atlas"
+          glyph="Cloud"
+          className={cx(subMenuContainerStyle, {
+            [subMenuActiveContainerStyle]: isCloud,
+          })}
+        >
+          <MenuItem href={urls?.mongoMenu?.cloud?.userPreferences}>
```

Comment:
These don't need the optionals anymore, right?

Reply:
👍